### PR TITLE
Restructuring of repo and introducing inspec-k8s

### PR DESCRIPTION
**Pull request by KonradSchieban**

### CODE REVIEW

**Comment by binamov** [comment]
Any chance to merge #9 in here and then update the README and walkthrough for the new approach?

File: run_profiles.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+inspec exec inspec-gke-cis-gcp -t gcp:// --input-file inputs.yml &
```
Comment (binamov) [inline run_profiles.sh:2]: If we're running processes in the bg then those processes should output a json for later review.

Comment (KonradSchieban) [reply run_profiles.sh:2]: agreed

**Comment by KonradSchieban** [comment]
> Any chance to merge #9 in here and then update the README and walkthrough for the new approach?

Suggest we merge #9 first into master, rebase and update the walkthrough after merging this PR.

File: inspec-gke-cis-ssh/controls/3.01-node-files.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  c = command('ps -ef | grep kube-proxy | grep -e "--kubeconfig=" | sed "s/^.*\(--kubeconfig=.* \).*$/\1/" | awk \'{print $1}\'').stdout.split('=')
+  kube_proxy_config_file_path = c[1].split("\n").first
+  
+  file_permissions = command("stat -c %a #{kube_proxy_config_file_path}").stdout.to_i
+
+  describe "[#{gcp_project_id}] File permissions of kube-proxy config file #{kube_proxy_config_file_path}" do
+    subject { file_permissions }
+    it 'should be 644 or more restrictive' do
+      expect(subject).to be_in [644, 640, 600, 400, 444, 440]
+    end
```
Comment (binamov) [inline inspec-gke-cis-ssh/controls/3.01-node-files.rb:55]: This should use the `file` resource with the `mode` property. Please note that octal notation requires this to be 0644 not just 644 to work.

File: inspec-gke-cis-ssh/controls/3.01-node-files.rb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  ref 'CIS Benchmark', url: cis_url.to_s
+  ref 'GCP Docs', url: 'https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/'
+
+  kubelet_config_file_path = command('ps -ef | grep kubelet | grep -e "--config " | sed "s/^.*\(--config .* \).*$/\1/"  | awk \'{print $2}\'').stdout.split("\n").first
+  file_permissions = command("stat -c %a #{kubelet_config_file_path}").stdout.to_i
+
+  describe "[#{gcp_project_id}] File permissions of kubelet config file #{kubelet_config_file_path}" do
+    subject { file_permissions }
+    it 'should be 644 or more restrictive' do
+      expect(subject).to be_in [644, 640, 600, 400, 444, 440]
```
Comment (binamov) [inline inspec-gke-cis-ssh/controls/3.01-node-files.rb:122]: same comment as line 55

File: inspec-gke-cis-ssh/controls/3.01-node-files.rb (hunk 2 above)
Comment (KonradSchieban) [reply inspec-gke-cis-ssh/controls/3.01-node-files.rb]: fixed with latest commit.

File: inspec-gke-cis-ssh/controls/3.01-node-files.rb (hunk 3 above)
Comment (KonradSchieban) [reply inspec-gke-cis-ssh/controls/3.01-node-files.rb]: fixed with latest commit.

**Comment by KonradSchieban** [comment]
Merging into dedicated branch for now

**Pull request merged**